quietly ignore worker requests for input from queues that don't exist

beerriot opened the pull request:
Fix for https://issues.basho.com/show_bug.cgi?id=1153

The likely cause of BZ1153 is that the vnode received a `'DOWN'` for the fitting process, and deleted the queue for that fitting _after_ the worker for that fitting had already sent its next_input request.  This led the vnode to pick up a next_input request that pointed to a queue it no longer had.

An alternate fix might be to hang onto the queue, but mark it as dead, until the exit for the worker is received.

beerriot committed:
quietly ignore worker requests for input from queues that don't exist

this may happen if a next-input request is already sitting in the
vnode's mailbox when the vnode decides to kill that worker as the result
of the worker's fitting exiting

rustyio commented:
Should we handle the same case in `archive_fitting/2`? It gets called in the same way as `next_input_internal/2`, except during handoff, so will likely have the same problem. See code snippet below...

``` erlang
archive_fitting(F, State) ->
    {ok, W} = worker_by_fitting(F, State), %% <--- This seems problematic.
    send_archive(W),
    CleanState = remove_worker(W, State),
    CleanState#state{workers_archiving=[W#worker{state={working, archive}}
                                        |State#state.workers_archiving]}.
```

beerriot committed:
ignore non-existent-queue worker requests in handoff as well (thanks Rusty)

Just as with the last commit (d40138f), there may be no queue available
for this fitting (for example, because the fitting died, and the queue
was cleaned up).  It should be save to ignore this request.

beerriot commented:
Good catch, Rusty.  I've just added a patch for that case as well.  Thanks!
